### ISSUE TITLE
Fix CssSyntaxError for escaped mixin name

### ISSUE THREAD

**Issue by keith0305** [stylelint/stylelint#4740]
> Clearly describe the bug

Stylelint is not able to recognize escaped mixin name - when \ occurs right after . (dot).

> Which rule, if any, is the bug related to?

`CssSyntaxError`

> What code is needed to reproduce the bug?
```less
.test-mixin() {
	padding-top: 1rem;
}

// totally valid mixin in LESS.
.test-\!mixin2() {
	padding-bottom: 1rem;
}

// also a totally valid mixin.
.\!test-mixin3() {
	padding-right: 1rem;
}

.btn-primary {
	.test-mixin();
	.test-\!mixin2(); // 👌 no issue
	.\!test-mixin3(); // CssSyntaxError here
}
```

> What stylelint configuration is needed to reproduce the bug?

Any rules

> Which version of stylelint are you using?

`13.3.3`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API with gulp

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's a LESS-specific syntax.

> What did you expect to happen?

Stylelint able to parse the LESS file.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
 18:2  ×  At-rule without name   CssSyntaxError
```

**Comment by jeddy3** [comment]
@keith0305 Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIB0YA2BDAzpgBMAOgHY45L74CEAtgJYAeNhAFAJQDcRAviADQiYBPQgBd0dBCFQxsvcBEIAzGgHMJBYjnwgATgFcpmLfDxESJLemEBaPVKuEIV3YQDWDgO6EjOYXpg9TMy0AI1QIMBd7RxgqAAdhAW9fXX9A8zkw7SirRgA3dFQaKCsACxg6JL8AjXTIKioYEWyY+MSEHyq0zRBYNHRtSxp5K1DwyIcrKF1YwrBLGCtY7QhYmG1hGmlvAG0ukmTUmrN1M1PulQdtGB29s61IQkwYMF0N3IWpmZo54WkrdxowhKkxoCgUa0a1nyqBShhAtxIAF1bpwuojqqctL0MAMNsNRhFspgShB1iV0IRiksVmsElYIO9tNoilt2gcMUEQAp5NYFOhaKgBNlPrN5vZ+azjOyulpuU0+QKhRNaNhGMorMpGmtvlYFTRBVYXDABO5SVBKikOekFM4wHjCFY5qgwNk8gUihbDpiubb7Y6Ci6Js5MLF0GAYMUabjSZ6rd0bYQ7UMHYVCDB+hqBlBNk0Jg5HqJKf1itmrkn5LGZSAjQIFAMGpNnjjBsMJjQ4qTC8JK0ctA1s+hdenhLoruKG0HXB4vGzOr2QBNYJhw5S1VYQ89Qd9AW0pXPvYvpqLflZLDkO+sbPpJR1LVXD18fgsnlI7aS4Xu7-OJi06ZgIKO4Y9ge0R0L4g5PLQkBhI8wGcm2hDQkUkwAaEz4YMSjoQPUkIfreXqctSqzrEqjjOG4ECeHB6Qvs8wikosTy6FAjh9Ng2TkdO1HdLRb5ZLETEsVYMBSA0uZkVOlEzp+BE0SJdEMQkqwcZJVGzl+3qYL4a55jA7iptc6myd0ziAipFFqTJcYWNYtgLL+QoGSMMDclcNxHOkBTuOgAhwnGJAnGc6TlOG8TuUF3oEi4BiYKeCi-FkmASuOCxRTFWj+XcXI0NoWn2NIvzmvCHlmOiCLnMolyGcYuwld66DxWsWE4SIWjlciJWokcZXzmedkpf6TzeFoYTuGsGVVn114DSGYYLA1CXDSAXk+bFmBqnYBkTb1tnTZB3wQDB+X6Yw82NdoS0rb522RWEhJoBA62EOqwQDOGwlxHSTnBC5pLVd0aaMjdnJRY6YRPS9b0LGmJ1pnFi3tBYqDeddvBVqDD0QyMUPHd9v1uYjy3I6tVhUPoGxWFtaPzhj4Nrq9c3rqG70-a5-1Iyja0bQsVPWSAoM0oQ9M4zDTkLeNhNXbFZOoBTvPo3dkSC8LjOze94sXZLxO+eu3OU6dwPpALqxC892Oq8zqX4+zROc4b3TQQxZR0INNujRLfOO1kzuU40yhAktxKdvb9zYWJ1gOfrcOs394VZVL9sBeVWghTAYXtLVEWckouXWGmWkRm1dU4D1WdaBcscZ+VNEJFIqbWHUVAUlAcLtSiaKe2HkL-CUgLSJbOSPCyl3a35VYvFp2GLMsxF0pHeNs3H9Wj4nJjFyndChd2VfF5552Orok9UNPUYJKv3o53l+eFUXWelxF5eVZXNXV90GvNeH5+cow62wLrz2bVOiMRWt8IodSCl1U4990jYn6C2B0wQKTqjVmdBGxgtCAw9lWWBuJkwjCQUzRmMcCboNtqtEOPQmxwL9KDfajsHSiyARrEenNSbkxoFHQyfMcHwOAWMdcMQDpHRQfDCWpCpb-2UIAtMFCeE0MVgIqCh1hgiOITbTBmtuFUNwfiBRENpELCbi7ORyY8IAEYtF9B0QghR4F9Rrjoco6Sb8V7U29CY4Y9DcZMPOiwkmMs5YGzcZyDxDovEiOYVrVh+ieZBMsc2P04SB5qKWho2R2jeHzyASkneQUObkOCacQKeSQCp3Ti-Xeb996hK-taHKV8CqF2KnfTK6QK4kJwJnLOVSEof0hLU84Q8-4xM4XwiIoCgrgLOJA0qfMEzljCWHQcjC4aRPEaPNhssOHy3nPMxJSzCHq18VEkmIydkX19HgxuEFknW1STAIGhT4yXOGEYj6rROF4QAAxzJeQw5KcwhqE3dpoqsey8GhiuCIMoTxYorIWD-YeJydYBO2XEsFfzFj9EhDCv4IjEWwHuYySRBiKHguGO4XuvwInHPWXbJ55dKSQngd4AATHzKQz0gRWAAF5rEcJOQE3E+xiHeV9U65i+b9hoIOcElhRwLCSYzNZLj6VSojDKoc8qxxKpZncwmaSGUgGlbK4cCqBqArdpRLB84TVapHGOSFOLpB4oHgS9RDybXejtXKh1CwBjPQWFGSwDEaVoNVQU9VA57XmoDZqekxEQ2JVuYvZFY9bUasHAARxSNoRyNA8rXO8as2lEaUXsM4RQu1Oa1j5sLQcsNYiy1cwAbEmRRrq25rrQ3BtKbn4A09aC780RDo9z7ig4VC5HAqixo0SMCh1wAW0O9eFk7CC6CoD9LIUh0DZjNny5YfjUZ83XZupqEw7GpnVAex6k6iK0iFJapaIKKH9SydHfVFSSkSNRZWp5Sd16lM3mnbeX7umynqXnRpRV2qtIqlVJeD9lr70bv05pYD27dT5rxeiWRLDaWCK8VKb0jTCDWm6oZHrHnYfknxU8wgCNEYTWsJNhzUFNowYOihOGGL4eZIRk8wbcNsecqm0hhqaOvmE3UYIjBWONs1nSyNVYeNZBk3J4TqjP3Nu47R4TS4VxQApD2zdGmGJ5hUZbSdqn9b1vqMsvSYtS35OPSpvTDFCh2abiJnJ4muNGps2899N4fluakwxASMBmKsQwrFJ9wLrVDs0u5-igkYtYFik66FLryOM3dUSr1nIbORei8JUS3d4ukJfQFlLjEotCXkuHLCR0HCiHtEtFiug0K6fC1kJSiqsBWrGklzkpkG6DefYlih0IUiGmNKabQxRKvdGq3zGbsSvMOdhuxxTzbNmBPbWtgos3PMme8wpo9LapFtq4VWdbtmzs3KIdpzj1G7vHY29YILn183528F8wIXVuCcCAA) with:

```less
.class {
  .\!mixin();
}
```

It's likely an issue with [the parser](https://github.com/shellscape/postcss-less) stylelint uses [to support Less](https://stylelint.io/about/syntaxes). 

Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time, so that it supports your use case of escaped mixins.

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

Please consider fixing the issue upstream as the syntax is no longer bundled by Stylelint.
